modules/carto/src/layers/vector-tile-layer.ts

```diff
@@ -134,11 +134,19 @@ export default class VectorTileLayer<
     const tileBbox = props.tile.bbox as any;
     const {west, south, east, north} = tileBbox;
 
+    const clipProps = {
+      extensions: [new ClipExtension(), ...(props.extensions || [])],
+      clipBounds: [west, south, east, north]
+    };
+
     const subLayerProps = {
       ...props,
       autoHighlight: false,
-      extensions: [new ClipExtension(), ...(props.extensions || [])],
-      clipBounds: [west, south, east, north]
+      _subLayerProps: {
+        'polygons-fill': clipProps,
+        'polygons-stroke': clipProps,
+        linestrings: clipProps
+      }
```

Comment:
It could be useful to include a comment linking to the issue, so we remember why point layers aren't included here.